src/jestMatchers.ts

```diff
@@ -0,0 +1,338 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import assert from 'assert';
+import type { MetadataBearer } from '@aws-sdk/types';
+import type { AwsCommand, AwsStub } from './awsClientStub';
+
+/**
+ * Provides {@link jest} matcher for testing {@link AwsStub} command calls
+ *
+ * @example
+ *
+ * ```ts
+ * import { mockClient } from "aws-sdk-client-mock";
+ * import { ScanCommand } from "@aws-sdk/client-dynamodb";
+ *
+ * const awsMock = mockClient(DynamoDBClient);
+ *
+ * awsMock.on(ScanCommand).resolves({
+ *   Items: [{ Info: { S: '{ "val": "info" }' }, LockID: { S: "fooId" } }],
+ * });
+ *
+ * it("Should call scan command", async () => {
+ *    // check result ... maybe :)
+ *    await expect(sut()).resolves.toEqual({ ... });
+ *
+ *    // Assert awsMock to have recevied a Scan Command at least one time
+ *    expect(awsMock).toHaveReceivedCommand(ScanCommand);
+ * });
+ * ```
+ */
+export interface AwsSdkJestMockMatchers<E, R> extends Record<string, any> {
```

Comment:
Let's add shorter aliases similarly to the ones in core jest matchers:

- `toReceiveCommandTimes`
- `toReceiveCommand`
- `toReceiveCommandWith`
- `toReceiveNthCommandWith`